arangod/Pregel/Conductor/LoadingState.cpp

```diff
@@ -0,0 +1,58 @@
+#include "LoadingState.h"
+
+#include "Pregel/Algorithm.h"
+#include "Pregel/Conductor.h"
+#include "Metrics/Gauge.h"
+#include "Pregel/Conductor/State.h"
+#include "Pregel/PregelFeature.h"
+#include "Pregel/WorkerConductorMessages.h"
+
+using namespace arangodb::pregel::conductor;
+
+#define LOG_PREGEL(logId, level)          \
+  LOG_TOPIC(logId, level, Logger::PREGEL) \
+      << "[job " << conductor._executionNumber << "] "
+
+Loading::Loading(Conductor& conductor) : conductor{conductor} {
+  conductor.updateState(ExecutionState::LOADING);
+  conductor._timing.loading.start();
+  conductor._feature.metrics()->pregelConductorsLoadingNumber->fetch_add(1);
+}
+
+Loading::~Loading() {
+  conductor._timing.loading.finish();
+  conductor._feature.metrics()->pregelConductorsLoadingNumber->fetch_sub(1);
+}
+
+auto Loading::run() -> void {
+  LOG_PREGEL("3a255", DEBUG) << "Telling workers to load the data";
+  auto res =
+      conductor._initializeWorkers(Utils::startExecutionPath, VPackSlice());
+  if (res != TRI_ERROR_NO_ERROR) {
+    LOG_PREGEL("30171", ERR) << "Not all DBServers started the execution";
+    // TODO change state to Canceled
+    conductor.changeState(StateType::Placeholder);
+  }
+}
+
+auto Loading::receive(Message const& message) -> void {
+  if (message.type() != MessageType::GraphLoaded) {
+    LOG_PREGEL("14df4", WARN)
+        << "When loading, we expect a GraphLoaded message, nothing else";
```

Comment:
Maybe we want to say here which message we received? that might ease debugging quite a bit.